error-prone-contrib/src/test/resources/tech/picnic/errorprone/refasterrules/ComparatorRulesTestInput.java

```diff
@@ -31,6 +31,7 @@ ImmutableSet<Comparator<String>> testNaturalOrder() {
         String::compareTo,
         Comparator.comparing(identity()),
         Comparator.comparing(s -> s),
+        Comparator.comparing(s -> 0),
```

Comment:
For each rule, added a non-identity transformation test case, to validate that the `@Matcher` annotation is present.